File: BOJ/DP/12101_1, 2, 3 더하기 2.py
import sys

def input(): return sys.stdin.readline().rstrip()

N, K = map(int, input().split())
DP = [[] for _ in range(12)]

DP[1] = ["1"]
DP[2] = ["1+1", "2"]
DP[3] = ["1+2", "1+1+1", "2+1", "3"]

for idx in range(4,N+1):
    for gap in range(3,0,-1):
        for previous in DP[idx-gap]:
            DP[idx].append(f"{previous}+{str(gap)}")
            
if len(DP[N]) < K: 
    print(-1)
else:
    answer = sorted(DP[N])
    print(answer[K-1])

File: BOJ/DP/12101_1, 2, 3 더하기 2.kt
fun main(){
    val (N, K) = readln().split(" ")
        .map{ it.toInt() }
    
    val DP = Array(12){ mutableListOf<String>() }
    DP[1] = mutableListOf("1")
    DP[2] = mutableListOf("1+1", "2")
    DP[3] = mutableListOf("1+1+1", "2+1", "1+2", "3")
    
    for(idx in 4..N) {
        for(gap in 1..3) {
            for(previous in DP[idx-gap]) {
                DP[idx].add("${previous}+${gap}")
            }
        }
    }
    
    if(DP[N].size < K) {
        print("-1")
    } else {
        print(DP[N].sorted()[K-1])
    }
}